packages/datetime/src/datePickerCaption.tsx

```diff
@@ -44,73 +47,51 @@ export class DatePickerCaption extends React.PureComponent<IDatePickerCaptionPro
         const startMonth = displayYear === minYear ? minDate.getMonth() : 0;
         const endMonth = displayYear === maxYear ? maxDate.getMonth() + 1 : undefined;
         const monthOptionElements = months
-            .map((name, i) => {
-                return (
-                    <option key={i} value={i.toString()}>
-                        {name}
-                    </option>
-                );
-            })
+            .map<IOptionProps>((month, i) => ({ label: month, value: i }))
             .slice(startMonth, endMonth);
 
-        const years = [minYear];
+        const years: Array<number | IOptionProps> = [minYear];
         for (let year = minYear + 1; year <= maxYear; ++year) {
             years.push(year);
         }
-        const yearOptionElements = years.map((year, i) => {
-            return (
-                <option key={i} value={year.toString()}>
-                    {year}
-                </option>
-            );
-        });
         // allow out-of-bounds years but disable the option. this handles the Dec 2016 case in #391.
         if (displayYear > maxYear) {
-            yearOptionElements.push(
-                <option key="next" disabled={true} value={displayYear.toString()}>
-                    {displayYear}
-                </option>,
-            );
+            years.push({ value: displayYear, disabled: true });
         }
 
         this.displayedMonthText = months[displayMonth];
 
         const monthSelect = (
-            <div className={Classes.DATEPICKER_CAPTION_SELECT} key="month">
-                <select
-                    className={Classes.DATEPICKER_MONTH_SELECT}
-                    onChange={this.handleMonthSelectChange}
-                    value={displayMonth.toString()}
-                >
-                    {monthOptionElements}
-                </select>
-                <Icon
-                    className={Classes.DATEPICKER_CAPTION_CARET}
-                    icon="caret-down"
-                    style={{ left: this.state.monthWidth + 5 }}
-                />
-            </div>
+            <HTMLSelect
+                iconProps={{ style: { left: this.state.monthWidth } }}
```

Comment:
Interesting. What's going on here? An icon is absolutely positioned inside the `<select>`? Oh, or is this referring to the caret? Can we just let the select morph to the width of the widest `<option>`, or does our style customization make that tricky?

__EDIT:__ Took a look at the GIFs. Looks odd to me when the caret is right next to the text but the hoverable area goes way beyond the icon to the right. Can we just follow the style of default `<select>`s and leave the caret flush on the right side of the hoverable region?

Reply:
@cmslewis i've got back and forth on this myself. you can disable the `left` property in the browser. i think it looks weirder with the two icons on either side of the year.

![image](https://user-images.githubusercontent.com/464822/44047423-24296360-9ee3-11e8-9509-a3f9c1a95c7c.png)